mergify_engine/worker.py

```diff
@@ -1160,6 +1184,9 @@ def get_shared_worker_ids(self) -> typing.List[int]:
     async def start(self) -> None:
         self._stopping.clear()
 
+        await ping_redis(self._redis_links.stream, "Stream")
+        await ping_redis(self._redis_links.cache, "Cache")
+
```

Comment:
what if redis crashes right here?
is the following `load_scripts` going to retry too?

Reply:
Yes: and if you retry load_scripts "for ever" you basically have a better version of "ping redis" IMHO.

It's probably a bit orthogonal to this PR, but handling script loading dynamically might also be worth it.

Basically, making sure that if you restart Redis *at any point in time* Mergify falls back on its feet is a good way to think about the problem.

Reply:
We were doing `load_script()` (note no ending 's') dynamically before but when `eval_sha()` is called from a pipeline it always succeeds and then an unrelated error is raised later when the pipeline is executed, making it hard to know which one is missing.

This is why we added this "load_scripts" to preload them on startup.

Maybe we can leverage `Client.on_connect()` to preload/reload them instead?

For migration, maybe we can just drop them for now, as all on-prem has run them.

I'm still in favor of pinging Redis on startup, as this improves the on-premise first installation experience.